src/main/java/net/helix/pendulum/crypto/Merkle.java

```diff
@@ -55,6 +60,44 @@
     }
 
 
+    public static List<byte[]> buildMerkleTransactionTree(List<Hash> leaves, Hash milestoneHash, Hash address){
+        if (leaves.isEmpty()) {
+            leaves.add(Hash.NULL_HASH);
```

Comment:
There were some concerns from Oliver about using `NULL_HASH`. Is it kosher here or we should use another filler?

Reply:
Then I will replace it wht List<List<Hash>>
Regarding NULL_HASH it can be replaced with other hash (but should exists in tangle), here it means that no tips are found, so their merkle root is NULL_HASH, in tangle it will look like this milestone is directly connected to genesis.

Reply:
>  in tangle it will look like this milestone is directly connected to genesis
why I don't think it is kosher.

Reply:
using [this hash](https://github.com/HelixNetwork/pendulum/blob/dev/src/main/java/net/helix/pendulum/conf/BasePendulumConfig.java#L1039) should be fine.

Reply:
> can be replaced with other hash (but should exists in tangle)
yes it has to be a valid tx_hash, not sure how, without implementing a(nother) genesis tx.